File: app/src/main/java/com/example/bangabandhuplay/ui/home/HomeSliderAdapter.java
package com.example.bangabandhuplay.ui.home;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.bangabandhuplay.R;
import com.example.bangabandhuplay.data.model.sliders.Original;
import com.example.bangabandhuplay.databinding.HomeSliderLayoutBinding;
import com.squareup.picasso.Picasso;
import java.util.List;

public class HomeSliderAdapter extends RecyclerView.Adapter<HomeSliderAdapter.mViewHolder> {

    private final List<Original> images;
    public HomeSliderAdapter(List<Original> images) {
        this.images = images;
    }

    @NonNull
    @Override
    public mViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new mViewHolder(HomeSliderLayoutBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull HomeSliderAdapter.mViewHolder holder, int position) {
        Original current = images.get(position);
        Picasso.get().load(current.getImage()).resizeDimen(R.dimen.slider_image_width, R.dimen.slider_image_height).into(holder.binding.sliderThumbnailIv);
    }

    @Override
    public int getItemCount() {
        return images != null ? images.size(): 0;
    }

    static class mViewHolder extends RecyclerView.ViewHolder{
        HomeSliderLayoutBinding binding;
        public mViewHolder(@NonNull HomeSliderLayoutBinding itemView) {
            super(itemView.getRoot());
            binding = itemView;
        }
    }
}

File: app/src/main/java/live/durbar/bangabandhuapp/ui/home/HomeSliderAdapter.kt
package live.durbar.bangabandhuapp.ui.home

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import live.durbar.bangabandhuapp.R
import live.durbar.bangabandhuapp.data.model.sliders.Original
import live.durbar.bangabandhuapp.databinding.HomeSliderLayoutBinding
import live.durbar.bangabandhuapp.ui.player.PlayerActivity
import live.durbar.bangabandhuapp.utils.Constants
import com.squareup.picasso.Picasso

class HomeSliderAdapter(private val images: List<Original>, private val context: Context) :
    RecyclerView.Adapter<HomeSliderAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            HomeSliderLayoutBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val current = images[position]
        val image = current.image
        val uuid = current.contentUrl
        val title = current.title

        image?.let {
            Picasso.get()
                .load(it)
                .resizeDimen(R.dimen.slider_image_width, R.dimen.slider_image_height)
                .into(holder.binding.sliderThumbnailIv)
        }

        holder.binding.root.setOnClickListener {
            if (uuid != null && uuid.isNotEmpty()) {
                context.startActivity(
                    Intent(context, PlayerActivity::class.java)
                        .putExtra(Constants.CONTENT_UUID, uuid)
                        .putExtra(Constants.CONTENT_SECTION_TITLE, "Slider")
                )
            }
            Constants.IS_MORE_CONTENT = false
            Constants.IS_MORE_HOME = false
        }
    }

    override fun getItemCount(): Int {
        return images.size
    }

    inner class ViewHolder(val binding: HomeSliderLayoutBinding) :
        RecyclerView.ViewHolder(binding.root)
}
